plugins/woocommerce-blocks/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md

```diff
@@ -108,29 +108,273 @@ We also have individual features or code blocks behind a feature flag, this is a
+
+This documentation will be amended following a review of this action.
+
+#### `experimental__woocommerce_blocks-checkout-submit`
+
+Fired when the checkout form is submitted.
+
+##### Args
+
+This action has no arguments.
+
+##### Example
+
+```js
+wp.hooks.addAction(
+	'experimental__woocommerce_blocks-checkout-submit',
+	'plugin/namespace',
+	() => {
+		console.log( 'The checkout form was submitted.' );
+	}
+);
+```
+
+#### `experimental__woocommerce_blocks-checkout-set-selected-shipping-rate
+
+Fired when a shipping rate is chosen on checkout.
+
+##### Args
+
+| Argument   | Type     | Description                    |
+|------------|----------|--------------------------------|
+| `shippingRateId`  | `string` | The selected shipping rate ID. |
+
+##### Example
+
+```js
+wp.hooks.addAction(
+	'experimental__woocommerce_blocks-checkout-set-selected-shipping-rate',
+	'plugin/namespace',
+	( { shippingRateId } ) => {
+		console.log( `Selected shipping rate was changed to ${ shippingRateId }` );
+	}
+);
+```
+
+#### `experimental__woocommerce_blocks-checkout-set-active-payment-method`
+
+Fired when a payment method is chosen on checkout.
+
+##### Args
+
+| Argument   | Type     | Description                      |
+|------------|----------|----------------------------------|
+| `paymentMethodSlug`  | `string` | The payment method slug.         |
+
+##### Example
+
+```js
+wp.hooks.addAction(
+	'experimental__woocommerce_blocks-checkout-set-active-payment-method',
+	'plugin/namespace',
+	( { paymentMethodSlug } ) => {
+		console.log( `The selected payment method was changed to ${ paymentMethodSlug }` );
+	}
+);
+```
+
+#### `experimental__woocommerce_blocks-checkout-render-checkout-form`
+
+Fired when the checkout form is rendered.
+
+##### Args
+
+This action has no arguments.
+
+##### Example
+
+```js
+wp.hooks.addAction(
+	'experimental__woocommerce_blocks-checkout-render-checkout-form',
+	'plugin/namespace',
+	() => {
+		console.log( 'The checkout form was rendered.' );
+	}
+);
+```
+
+#### `experimental__woocommerce_blocks-checkout-set-email-address`
+
+Fired when an email address is added during checkout.
+
+##### Args
+
+This action has no arguments.
+
+##### Example
+
+```js
+wp.hooks.addAction(
+	'experimental__woocommerce_blocks-checkout-set-email-address',
+	'plugin/namespace',
+	( { product, quantity } ) => {
+		console.log( 'The email address was changed.' );
+	}
+);
+```
+
+#### `experimental__woocommerce_blocks-checkout-set-shipping-address`
+
+Fired when a shipping address is added during checkout.
+
+##### Args
+
+This action has no arguments.
+
+##### Example
+
+```js
+wp.hooks.addAction(
+	'experimental__woocommerce_blocks-cart-set-shipping-address',
+	'plugin/namespace',
+	( { product, quantity } ) => {
+		console.log( 'The shipping address was changed.' );
+	}
+);
+```
+
+#### `experimental__woocommerce_blocks-checkout-set-billing-address`
+
+Fired when a billing address is added during checkout.
```

Comment:
Again maybe if its whenever it's changed the description should explicitly say that for clarity?